internal/collect_es6_sources.bzl

```diff
@@ -29,15 +29,24 @@ def collect_es6_sources(ctx):
   """
 
   non_rerooted_files = depset()
+  rerooted_files = []
   for dep in ctx.attr.deps:
     if hasattr(dep, "typescript"):
-      non_rerooted_files = depset(transitive = [non_rerooted_files, dep.typescript.transitive_es6_sources])
+      non_rerooted_files += depset(transitive = [non_rerooted_files, dep.typescript.transitive_es6_sources]).to_list()
+    elif hasattr(dep, "closure_js_library"):
+      non_rerooted_files += dep.closure_js_library.srcs
+    elif hasattr(dep, "files"):
+      non_rerooted_files += dep.files
+    else:
+      fail(
+          ("%s is neither a TypeScript nor a Closure JS library producing rule." % dep.label) +
+          "\nDependencies must be ts_library, ts_declaration, or closure_js_library.") 
 
-  rerooted_files = []
   for file in non_rerooted_files.to_list():
     rerooted_file = ctx.actions.declare_file(
-      "%s.es6/%s" % (
+      "%s.es6/node_modules/%s/%s" % (
         ctx.label.name,
+        ctx.workspace_name,
```

Comment:
shouldn't it be ctx.label.workspace_root? what if the file comes from an external workspace?

Reply:
That already works (I tested it with my angularjs demo is rules_closure)

Reply:
hmm, I'm not convinced, seems like it should match `file`